cmd/libs/go2idl/client-gen/README.md

```diff
@@ -0,0 +1,23 @@
+Client-gen is an automatic tool that generates [clientset](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/client-package-structure.md#high-level-client-sets) based on API types. This doc introduces the use the client-gen, and the release cycle of the generated clientsets.
+
+## Using client-gen
+
+The workflow include four steps:
+- Marking API types with tags: in `pkg/apis/${GROUP}/${VERSION}/types.go`, mark the types (e.g., Pods) that you want to generate clients for with the `// +genclient=true` tag.  If the resource associated with the type is not namespace scoped (e.g., PersistentVolume), you need to append the `nonNamespaced=true` tag as well.
```

Comment:
Hm. Maybe. A root and a group/version filter may go a long way.

On Fri, Feb 19, 2016 at 3:29 PM, Chao Xu <notifications@github.com> wrote:

> In cmd/libs/go2idl/client-gen/README.md
> <https://github.com/kubernetes/kubernetes/pull/21436#discussion_r53533658>
> :
>
> > @@ -0,0 +1,23 @@
> > +Client-gen is an automatic tool that generates [clientset](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/client-package-structure.md#high-level-client-sets) based on API types. This doc introduces the use the client-gen, and the release cycle of the generated clientsets.
> > +
> > +## Using client-gen
> > +
> > +The workflow include four steps:
> > +- Marking API types with tags: in `pkg/apis/${GROUP}/${VERSION}/types.go`, mark the types (e.g., Pods) that you want to generate clients for with the `// +genclient=true` tag.  If the resource associated with the type is not namespace scoped (e.g., PersistentVolume), you need to append the `nonNamespaced=true` tag as well.
>
> For third party users, they may not want to generate clients for every
> group/version in the root, and the directory structure can be irregular, so
> I think we need more flexible syntax like a list of pairs of <path to
> types.go, group/version>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/21436/files#r53533658>.
>